config/src/main/java/com/alibaba/nacos/config/server/service/ConfigOperationService.java

```diff
@@ -130,8 +129,17 @@ public Boolean publishConfig(ConfigForm configForm, ConfigRequestInfo configRequ
                         "Cas publish fail, server md5 may have changed.");
             }
         } else {
-            configOperateResult = configInfoPersistService.insertOrUpdate(configRequestInfo.getSrcIp(),
-                    configForm.getSrcUser(), configInfo, configAdvanceInfo);
+            if (configRequestInfo.getUpdateForExist()) {
+                configOperateResult = configInfoPersistService.insertOrUpdate(configRequestInfo.getSrcIp(),
+                        configForm.getSrcUser(), configInfo, configAdvanceInfo);
+            } else {
+                try {
+                    configOperateResult = configInfoPersistService.addConfigInfo(configRequestInfo.getSrcIp(),
+                            configForm.getSrcUser(), configInfo, configAdvanceInfo);
+                } catch (DataIntegrityViolationException ive) {
+                    return false;
```

Comment:
declare another exception here of ConfigAlreadyExistsException for a existed config when updateForExists being set to false.